src/main/kotlin/org/rust/lang/utils/RsDiagnostic.kt

```diff
@@ -864,6 +866,23 @@ sealed class RsDiagnostic(
         }
     }
 
+    class ImplForNonAdtError(
+        element: PsiElement,
+        private val impl: RsImplItem,
+        private val type: Ty
+    ) : RsDiagnostic(element) {
+        override fun prepare() = PreparedAnnotation(
+            ERROR,
+            E0118,
+            "Can impl only `struct`s, `enum`s and `union`s (and primitives in `no_core` crates)",
+            fixes = if (type is TyPrimitive && NO_CORE.availability(element) == FeatureAvailability.CAN_BE_ADDED) {
+                listOf(NO_CORE.addFeatureFix(element))
```

Comment:
No-core is not only feature, but crate attribute `#[no_core]`, and I also think that this fix isn't reasonable